### PR TITLE
Remove gke storage class yaml, add patching sc in deployer

### LINKED ISSUE #1630: GKE: volumeBindingMode is set to Immediate
With GKE we used to create a new storage class with the `volumeBindingMode` field set to `WaitForFirstConsumer` : https://github.com/elastic/cloud-on-k8s/blob/ee0645187e48e282a9af18425aa846c6fe6655ec/operators/hack/gke-cluster.sh#L90

It seems that #1594 removes this feature. This could lead to Pod that can't be scheduled in our e2e tests.

### PR DESCRIPTION
**Pull request by david-kow**
Fixes #1630.

### CODE REVIEW

**Comment by anyasabo** [comment]
jenkins test this please

File: hack/deployer/runner/gke.go [hunk 1]
```diff
@@ -220,6 +224,34 @@ func (d *GkeDriver) GetCredentials() error {
...
+func (d *GkeDriver) patchStorageClass() error {
+	log.Println("Patching storage class...")
+
+	if exists, err := NewCommand("kubectl get sc").OutputContainsAny("standard-customized"); err != nil {
+		return err
+	} else if exists {
+		return nil
+	}
+
+	sc, err := NewCommand("kubectl get sc standard -o yaml").Output()
```
Comment (pebrc) [inline hack/deployer/runner/gke.go]: We should not rely on naming but on the annotation `storageclass.beta.kubernetes.io/is-default-class=true` and `storageclass.kubernetes.io/is-default-class`

Comment (david-kow) [reply hack/deployer/runner/gke.go]: This was mirroring previous behavior (below). I've changed to use annotations now. https://github.com/elastic/cloud-on-k8s/blob/ee0645187e48e282a9af18425aa846c6fe6655ec/operators/hack/gke-cluster.sh#L91

File: hack/deployer/runner/gke.go [hunk 2]
```diff
@@ -220,6 +224,55 @@ func (d *GkeDriver) GetCredentials() error {
...
+	if defaultName == "" {
+		return fmt.Errorf("default storageclass not found")
+	}
+
+	sc, err := NewCommand(fmt.Sprintf("kubectl get sc %s -o yaml", defaultName)).Output()
+	if err != nil {
+		return err
+	}
+
+	sc = strings.Replace(sc, "name: standard", "name: standard-customized", -1)
```
Comment (pebrc) [inline hack/deployer/runner/gke.go]: this needs to use `defaultName`

**Pull request merged**